setup.py

```diff
@@ -14,11 +16,55 @@ def c_files_in(directory):
         paths.append(os.path.join(directory, f))
     return paths
 
+def send_html_entities(entities_file, outfh, seen_entities):
+    # Convert entity list from HTML5 spec JSON and send it to gperf
+    with open(entities_file) as entitiesfh:
+        for entity, entityinfo in sorted(json.load(entitiesfh).items()):
+            if not entity.endswith(';'):
+                continue
+            if entity in seen_entities:
+                continue
+            seen_entities.add(entity)
+            # Some sanity checks on the codepoints
+            assert len(entityinfo['codepoints']) <= 2
+            for bad_range in [xrange(0, 8), xrange(11, 12), xrange(14, 31),
+                              xrange(55296, 57343), xrange(65534, 65535)]:
+                for codepoint in entityinfo['codepoints']:
+                    assert codepoint not in bad_range
+            representation = ",".join(str(x) for x in entityinfo['codepoints'])
+            # Output the entity
+            outfh.write(entity + ", {" + representation + "}\n")
 
-def process_gperf_file(gperf_file, output_file):
+
+def process_gperf_file(gperf_file, entities_file, output_file, force=False):
     if not find_executable("gperf"):
         raise Exception("Couldn't find `gperf`, is it installed?")
-    subprocess.check_call(["gperf", gperf_file, "--output-file=%s" % output_file])
+
+    # Do not rerun gperf if no change to input files
+    if not force and not newer_group((gperf_file, entities_file), output_file):
+        return
+
+    # Output combined gperf input file
+    gperf_temp_file = gperf_file + ".generated"
+    seen_entities = set()
+    found_separator = 0
+    with open(gperf_temp_file, 'w') as outfh:
+        with open(gperf_file) as f:
+            for line in f:
+                entity = line.strip()
+                if entity == '%%':
+                    found_separator += 1
+                    if found_separator == 2:
```

Comment:
Nit: A comment explaning which number maps to which section would be helpful for people who don't know gperf well